atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/timestamp/InDbTimestampBoundStore.java

```diff
@@ -131,6 +131,7 @@ private InDbTimestampBoundStore(
         this.physicalBoundStoreStrategy = physicalBoundStoreStrategy;
     }
 
+    @SuppressWarnings("GuardedBy") // TODO (jkong): synchronize?
```

Comment:
This seems odd and scary to suppress. I think we want something like the following instead:

```
    private volatile DBType dbType;
    private DBType getDbType(Connection connection) {
        DBType type = this.dbType; // volatile read
        if (type == null) {
            synchronized (this) {
                type = this.dbType;
                if (type == null) {
                    type = ConnectionDbTypes.getDbType(connection);
                    this.dbType = type;
                }
            }
        }
        return type;
    }
```